phac-nml/irida-next
# Update: Policy manageable scopes

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates the `manageable` scopes for the project and namespace policies which will now return project namespaces and groups which the user has `modify` access to. A bug in the base project `relation` scope was fixed so that the correct linked projects are returned.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1) Create 2 new users (newuser1@email.com and newuser2@email.com)
2) Login as newuser1@email.com
3) Create a new group `Group A`, a project `Group A Project 1` under `Group A`, a subgroup `Group A Subgroup 1` under `Group A`, and a project `Group A Subgroup 1 Project 1`  under `Group A Subgroup 1`. For both projects created add a sample as well.
4) Logout and Login as newuser2@email.com
5) Create a new group `Group B`, a project `Group B Project 1` under `Group B`, a subgroup `Group B Subgroup 1` under `Group B`, and a project `Group B Subgroup 1 Project 1` under `Group B Subgroup 1`. For both projects created add a sample as well.
6) Visit the `Members` page for `Group B`
7) Invite group `Group A Subgroup 1` with group access level `Maintainer`
8) Logout and Login as newuser1@email.com 
9) On the groups page you should see `Group A` which is expandable to show `Group A Subgroup 1`, and `Group B` which is expandable to show `Group B Subgroup 1`
10) Go to the Create New Project page
11) In the `Project URL` you should see the following 5 options listed: newuser1's namespace, `Group A` and it's subgroup, `Group B` and its subgroup
12) Go to `Samples` page of the project created under `Group A` 
13) Select the sample and click the `Transfer Samples` button
14) You should see the projects which are under `Group A Subgroup 1`, `Group B`, and `Group B Subgroup 1`
15) Now Logout and Login as `newuser2@email.com` and go into the `Members` page for `Group B` and then click on the `Groups` tab. Update the group access level to `Analyst`
16) Logout and Login as `newuser1@email.com` and repeat steps 10 through 14. Verify that you can no longer see `Group B`, `Group B Subgroup 1` on the `Create New Project` page, and that you can no longer see projects under `Group B` and `Group B Subgroup 1` when transferring a sample in step 14


## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
